service/history/timerQueueProcessor.go

```diff
@@ -93,6 +104,13 @@ func (t *timerQueueProcessorImpl) SetCurrentTime(clusterName string, currentTime
 	standbyTimerProcessor.setCurrentTime(currentTime)
 }
 
+func (t *timerQueueProcessorImpl) FailoverDomain(domainID string, standbyClusterName string) {
+	// we should consider make the failover idempotent
+	failoverTimerProcessor := newTimerQueueFailoverProcessor(t.shard, t.historyService, domainID, standbyClusterName, t.logger)
```

Comment:
how is the life-cycle of this cursor managed?  Someone needs to call stop on it? or it calls stop by itself?

Reply:
timer ack manager will have a finished chan which used by timer active processor, when chan fired, active processor will stop it self